docs/src/components/Example.tsx

```diff
@@ -18,3 +19,32 @@ export function Example({ children, className = '' }: ExampleProps) {
     </Card>
   );
 }
+
+export function ExampleCode({ children }) {
+  const [copied, setCopied] = React.useState(false);
+  const [text, setText] = React.useState('');
+  const ref = React.useRef(null);
+
+  React.useLayoutEffect(() => {
+    setText(ref.current.innerText);
+    return () => {};
+  }, [children]);
```

Comment:
Is this super weird/bad code? I wanted to make it easy to add a code block that is passed to prism at the remark/rehype layer and still be able to copy its contents. So I'm adding a ref to a div wrapping the code block and setting its innerText to a state in a layout effect. It works. I don't know if there is a better way to do this...